Index: docker/build.sh
===================================================================
--- ./astrobee/src/astrobee/scripts/docker/build.sh
+++ ./astrobee/src/astrobee/scripts/docker/build.sh.new
@@ -66,15 +66,28 @@ elif [ "$os" = "focal" ]; then
   PYTHON='3'
 fi
 
+if [ "${NUM_JOBS}" = "" ]; then
+  NUM_JOBS=1
+else
+  NUM_JOBS=$NUM_JOBS
+fi
+
 echo "Building Ubuntu $UBUNTU_VERSION image"
 docker build ${rootdir}/ \
             -f ${rootdir}/scripts/docker/astrobee_base.Dockerfile \
+            --build-arg baseImage=${baseImage} \
             --build-arg UBUNTU_VERSION=${UBUNTU_VERSION} \
             --build-arg ROS_VERSION=${ROS_VERSION} \
             --build-arg PYTHON=${PYTHON} \
-            -t astrobee/astrobee:latest-base-ubuntu${UBUNTU_VERSION}
+            --build-arg HTTP_PROXY=${HTTP_PROXY} \
+            --build-arg HTTPS_PROXY=${HTTPS_PROXY} \
+            --build-arg http_proxy=${http_proxy} \
+            --build-arg https_proxy=${https_proxy} \
+            -t ${astrobeeBaseTag}
 docker build ${rootdir}/ \
             -f ${rootdir}/scripts/docker/astrobee.Dockerfile \
             --build-arg UBUNTU_VERSION=${UBUNTU_VERSION} \
             --build-arg ROS_VERSION=${ROS_VERSION} \
-            -t astrobee/astrobee:latest-ubuntu${UBUNTU_VERSION}
\ No newline at end of file
+            --build-arg NUM_JOBS=${NUM_JOBS} \
+            --build-arg TAG=${astrobeeBaseTag} \
+            -t ${astrobeeTag}
