Index: launch/sim.launch
===================================================================
--- ./src/astrobee/astrobee/launch/sim.launch
+++ ./src/astrobee/astrobee/launch/sim.launch.new
@@ -53,7 +53,7 @@
   <arg name="default_robot" default="true" />   <!-- Insert default robot    -->
   <arg if="$(eval arg('world')=='iss' and arg('perch')==0)"
        name="pose"
-       default="9.816 -9.806 4.293 0 0 0 1" />    <!-- iss: docked in JPM berth 1 --> 
+       default="10.7615 -6.8849 5.31647 0 0 -0.707 0.707"/> <!-- krpc3 start position -->
        <!-- default="9.817 -10.312 4.293 1 0 0 0" /> - iss: docked in JPM berth 2 -->
   <arg if="$(eval arg('world')=='iss' and arg('perch'))"
        name="pose"
@@ -93,6 +93,9 @@
   <!-- Declare our global logging format -->
   <env name="ROSCONSOLE_FORMAT" value="[${severity}] [${time}] : (${logger}) ${message}"/>
 
+  <!-- Set simulation parameter -->
+  <param name="krpc/on_simulation" value="true" />
+
   <!-- Always launch on the local machine -->
   <group>
     <machine name ="local" address="localhost" default="true"/>
@@ -233,4 +236,15 @@
   <!-- Default call initialize bias so config file is created if not already present -->
   <node pkg="rosservice" type="rosservice" name="imu_calibration" args="call --wait /gnc/ekf/init_bias" />
 
+  <node name="noise" pkg="topic_tools" type="relay" args="/gnc/ekf /gnc/ekf_noise" respawn="true" />
+  <node name="teleop_tool" pkg="executive" type="teleop_tool" args="-set_face_forward 'off'" />
+  <node name="marker_publisher" pkg="rostopic" type="rostopic"
+    args="pub --once /command ff_msgs/CommandStamped &quot;{
+                cmd_name: 'setOperatingLimits', subsys_name: 'Settings', args: [
+                {data_type: 5, s: iss_jaxa}, {data_type: 5, s: nominal},
+                {data_type: 2, f: 0.1500}, {data_type: 2, f: 0.0175},
+                {data_type: 2, f: 0.0873}, {data_type: 2, f:  0.1745},
+                {data_type: 2, f: 0}
+  ]
+  }&quot;" />
 </launch>
